src/puppetlabs/trapperkeeper/services/scheduler/scheduler_service.clj

```diff
@@ -19,6 +19,33 @@
       (tk/service-context)
       :jobs))
 
+(defn- queue-job!
+  ([this id]
+   (queue-job! this id {}))
+  ([this id opts]
+   (let [result (merge {:id id} opts)]
+    (swap! (get-jobs this) conj result)
+    result)))
+
+(defn- dequeue-job!
+  [this id]
+  (if-let [item (first (filter #(= id (:id %)) @(get-jobs this)))]
+    (swap! (get-jobs this) disj item)))
+
+(defn- wrap-remove-job-by-after-id!
```

Comment:
It took me a while to understand what was up with this, but I think I finally get it (mostly).  But I have a handful of comments:

* Most importantly, why do we need a separate `after-id`? It seems like we could reuse the existing ID for this ... but maybe I'm not understanding something about how `after-id` is used.

*  Does this mean that, before these changes, `after` job IDs were only added to the list of jobs, but never removed?

* Finally, I struggled with this name.  I think that, if this function had a docstring, a simple name like `after-job` might be sufficient.